### PR TITLE
MM-47310 : Migrated enterprise/permissions e2e tests to Typescript

### PR DESCRIPTION
**Pull request by cyberbuff**
#### Summary

Converted following spec files under /integration/enterprise/permissions to Typescript

1. team_permissions_spec.js

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21317

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]
Thanks @cyberbuff LGTM!

File: e2e/cypress/tests/integration/enterprise/permissions/team_permissions_spec.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ describe('Team Permissions', () => {
                 cy.apiAddUserToChannel(channel.id, testUser.id);
                 testPrivateCh = channel;
             });
-            cy.apiCreateUser().then(({user: newUser}) => {
+            cy.apiCreateUser({}).then(({user: newUser}) => {
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/enterprise/permissions/team_permissions_spec.ts:73]:
```suggestion
            cy.apiCreateUser().then(({user: newUser}) => {
```

**Review by furqanmlk — APPROVED** [review]
Thanks @cyberbuff 
LGTM !

**Pull request merged**